### PR TITLE
precommit: unify formatting with prettier +TPU

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

In particular it standardise the YAML structures

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18605.org.readthedocs.build/en/18605/

cc @borda @carmocca @justusschock @awaelchli

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -112,8 +112,15 @@ repos:
               README.md
           )$
 
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v3.0.3
+    hooks:
+      - id: prettier
+        # https://prettier.io/docs/en/options.html#print-width
+        args: ["--print-width=120"]
```
Comment (awaelchli) [inline .pre-commit-config.yaml:120]: Should we use https://prettier.io/docs/en/cli#--ignore-path instead of adding a new file to the root of the project?

Comment (Borda) [reply .pre-commit-config.yaml:120]: Good point, I was using extra file as we did in template (it was needed there to share ignore between precommit amd stands JS build tools)

Will update it...

Comment (Borda) [reply .pre-commit-config.yaml:120]: well trying it with `--ignore-path="**/*.md"` and seems not working at all

Comment (awaelchli) [reply .pre-commit-config.yaml:120]: It takes a path to an ignore file, not a pattern :)

**Comment by Borda** [comment]
it is annoying to be blocked because of TPU for no reason :(
```
Connection to 35.186.65.77 closed by remote host.
Retrying: SSH command error: [/usr/bin/ssh] exited with return code [255].
```

**Pull request merged**